lib/wadler/print.rb

```diff
@@ -93,7 +99,7 @@ def nest(indent, open_obj = '', close_obj = '',
     ensure
       @current_indent -= indent
 
-      if !close_obj.empty?
+      if !close_obj&.empty?
```

Comment:
Why don't use the straightforward `if close_obj`?  I very much doubt Booleans are accepted here anyway.  Keep it idiomatic.

Reply:
I mean if `close_obj.present?`, my bad.  But given the code, I see that anyway nil is invalid, so I don't understand why you try to support nil here.

Reply:
This portion of the code is present in the ensure block. If the method receives a nil close_obj and I don't check for nil, the program breaks (accessing nil element) because the ensure block will be executed after I raise the ArgumentError

Reply:
So maybe you should rather use a nested `begin/ensure/end` block inside the routine, and check preconditions **before**.

As it is, it looks clumsy to me.

Actually, anyway, move all precondition checks before any actual content (not even the `@current_indent thing`.

See what I mean?

Reply:
yes